### PR TITLE
Rename zenith to neon in python tests

### PR DESCRIPTION
**Pull request by bojanserafimov**
This will break continuity of perf reports since tests are renamed. Better early than later though.

We could mark tests with unique ids which will also allow us to explicitly break continuity without changing names, when the test significantly changes. But this seems too fancy for now.

### CODE REVIEW

File: test_runner/README.md [hunk 1]
```diff
@@ -51,8 +51,8 @@ Useful environment variables:
 should go.
 `TEST_SHARED_FIXTURES`: Try to re-use a single pageserver for all the tests.
 `ZENITH_PAGESERVER_OVERRIDES`: add a `;`-separated set of configs that will be passed as
-`--pageserver-config-override=${value}` parameter values when zenith cli is invoked
-`RUST_LOG`: logging configuration to pass into Zenith CLI
+`--pageserver-config-override=${value}` parameter values when neon cli is invoked
```
Comment (hlinnaka) [inline test_runner/README.md:54]:
```suggestion
`--pageserver-config-override=${value}` parameter values when neon_local cli is invoked
```

**Review by hlinnaka — APPROVED** [review]

**Comment by aome510** [comment]
Thanks for the work 💪 .

> [Revert envvar changes](https://github.com/neondatabase/neon/pull/1871/commits/3695c49e7358b1f1e6e51530b7cafc495e042c4a)

I think we only use `ZENITH_PAGESERVER_OVERRIDES` in tests (specifcally `test_remote_storage`), so it's safe to rename it to neon. 

Regarding `ZENITH_BIN`, the problem is that we initialize the circle CI files in https://github.com/neondatabase/neon/blob/main/.circleci/config.yml#L160-L167. The failed tests can be fixed by modifying those lines.

**Review by aome510 — APPROVED** [review]
LGTM beside a minor comment. Those can be addressed in a separate if you want.

**Review by phoenix24 — APPROVED** [review]

**Comment by aome510** [comment]
> This will break continuity of perf reports since tests are renamed. Better early than later though.

One thing I realized recently is that there are many legacy tests, e.g. `test_bulk_insert.py::test_bulk_insert`, which haven't been updated for a long time. Those tests are deprecated because we add `[vanilla]` and `[zenith]` suffixes.

**Pull request merged**